experiments/utils.py

```diff
@@ -344,19 +346,20 @@ class SessionUser(WebUser):
     def __init__(self, session, request=None):
         self.session = session
         self.request = request
+        self.redis = get_redis_client()
+        self.enrollments_key = "experiments:enrollments:%s" % self._participant_identifier()
+        self.goals_key = "experiments:goals:%s" % self._participant_identifier()
         super(SessionUser, self).__init__()
 
     def _get_enrollment(self, experiment):
-        enrollments = self.session.get('experiments_enrollments', None)
+        enrollments = self.redis.hgetall(self.enrollments_key)
         if enrollments and experiment.name in enrollments:
-            alternative, _, _, _ = _session_enrollment_latest_version(enrollments[experiment.name])
+            alternative, _, _, _ = _session_enrollment_latest_version(pickle.loads(enrollments[experiment.name]))
```

Comment:
Not looked at why we're using this, but pickle is notirious for security holes, can we use something else here?